tests/Aserto.AspNetCore.Middleware.Tests/AsertoCheckMiddlewareTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Aserto.AspNetCore.Middleware.Clients;
+using Aserto.AspNetCore.Middleware.Extensions;
+using Aserto.AspNetCore.Middleware.Options;
+using Aserto.AspNetCore.Middleware.Tests.Testing;
+using Google.Protobuf.WellKnownTypes;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Aserto.AspNetCore.Middleware.Tests
+{
+    public class AsertoCheckMiddlewareTests
+    {
+        [Fact]
+        public async Task AsertoCheckAllows()
+        {
+
+            var checkOptions = new CheckOptions();
+            checkOptions.BaseOptions = new AsertoOptions();
```

Comment:
I don't know if it's idiomatic in .NET but would it be possible to pass in all the values at construction time and make `AsertoOptions` immutable (with no property setters)?

Reply:
I think it might be possible, however we might have a couple of complication due to how the service extensions implementations are set at the moment, but I think we can do this in a future contribution. This PR already contains a lot of changes.